pkg/util/validation/limits.go

```diff
@@ -209,7 +209,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxGlobalExemplarsPerUser, "ingester.max-global-exemplars-per-user", 0, "The maximum number of exemplars in memory, across the cluster. 0 to disable exemplars ingestion.")
 	f.Var(&l.ActiveSeriesCustomTrackersConfig, "ingester.active-series-custom-trackers", "Additional active series metrics, matching the provided matchers. Matchers should be in form <name>:<matcher>, like 'foobar:{foo=\"bar\"}'. Multiple matchers can be provided either providing the flag multiple times or providing multiple semicolon-separated values to a single flag.")
 	f.Var(&l.OutOfOrderTimeWindow, "ingester.out-of-order-time-window", fmt.Sprintf("Non-zero value enables out-of-order support for most recent samples that are within the time window in relation to the TSDB's maximum time, i.e., within [db.maxTime-timeWindow, db.maxTime]). The ingester will need more memory as a factor of rate of out-of-order samples being ingested and the number of series that are getting out-of-order samples. If query falls into this window, cached results will use value from -%s option to specify TTL for resulting cache entry.", resultsCacheTTLForOutOfOrderWindowFlag))
-	f.BoolVar(&l.NativeHistogramsIngestionEnabled, "ingester.native-histograms-ingestion-enabled", false, "Enable ingestion of native histogram samples. If false, native histogram samples are ignored without an error.")
+	f.BoolVar(&l.NativeHistogramsIngestionEnabled, "ingester.native-histograms-ingestion-enabled", false, "Enable ingestion of native histogram samples. If true, -query-frontend.query-result-response-format must be set to 'protobuf' otherwise query sharding will fail. If false, native histogram samples are ignored without an error.")
```

Comment:
I understand we don't have a better place now, but documenting this at "ingestion-enabled" flag seems like a wrong place that many people will miss. I think we need an article describing the feature and how to enable it properly.

Reply:
:+1 on Peter's feedback. Moreover, couldn't we just bypass query sharding if native histograms is enabled and the response format != protobuf instead of letting it fail?

Reply:
I think bypassing automatically would just create confusion. Why are my queries suddenly slow? Then the person has to go into the documentation anyway.
I'd prefer to just get rid of the JSON option and remove the format flag altogether.

Reply:
cc @charleskorn

Reply:
Given we've made the protobuf format the default in #4557, I think this is less of an issue now. 

> Moreover, couldn't we just bypass query sharding if native histograms is enabled and the response format != protobuf instead of letting it fail?

Do we know when making a query that native histograms are enabled (and therefore know to bypass query sharding)?

Reply:
> Given we've made the protobuf format the default in https://github.com/grafana/mimir/pull/4557, I think this is less of an issue now.

True.

> Do we know when making a query that native histograms are enabled (and therefore know to bypass query sharding)?

Query-frontend already requires runtime config (because of query sharding). When `native_histograms_ingestion_enabled` is configured on a per-tenant level in the runtime config, then it's already available to query-frontend. Otherwise, we need to document that `-ingester.native-histograms-ingestion-enabled` (or global YAML setting) needs to be specified to query-frontend too.